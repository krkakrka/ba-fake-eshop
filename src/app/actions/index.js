import { RSAA } from 'redux-api-middleware';

const ADD_TO_CART = 'ADD_TO_CART';
const ADD_TO_FAVOURITES = 'ADD_TO_FAVOURITES';
const GET_PRODUCTS_START = 'GET_PRODUCTS_START';
const GOT_PRODUCTS = 'GOT_PRODUCTS';
const GET_PRODUCTS_ERROR = 'GET_PRODUCTS_ERROR';
const SET_CART_QUANTITY = 'SET_CART_QUANTITY';

const getProducts = () => {
  return (dispatch) => {
    dispatch({
      [RSAA]: {
        endpoint: 'https://blooming-cove-33093.herokuapp.com/food-shop/products',
        method: 'GET',
        types: [GET_PRODUCTS_START, GOT_PRODUCTS, GET_PRODUCTS_ERROR]
      }
    });
  };
};

export {
  ADD_TO_CART,
  ADD_TO_FAVOURITES,
  GET_PRODUCTS_START,
  GOT_PRODUCTS,
  GET_PRODUCTS_ERROR,
  SET_CART_QUANTITY,
  getProducts,
};
