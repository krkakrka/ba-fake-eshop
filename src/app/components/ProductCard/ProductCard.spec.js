import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import {
  MemoryRouter,
  BrowserRouter
} from "react-router-dom";
import { act } from "react-dom/test-utils";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { ProductCard } from './index';
import ContainerProductCard from './index';
import { SET_CART_QUANTITY } from '../../actions';

describe.only('ProductCard', () => {
  const mockProduct = { name: 'name', image: 'imgUrl', price: '1', currencySymbol: '$', id: 'prodId' }
  const mockLocation = { patname: '/root' };
  const mockCartProducts = { productToQuantity: { a: 5 } };
  let container = null;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should render correct snapshot', () => {
    act(() => {
      const el = <ProductCard
        product={mockProduct}
        location={mockLocation}
        cartProducts={mockCartProducts}
      />;
      render(el, container);
    })
    expect(container.innerHTML).toMatchSnapshot();
  });
});
