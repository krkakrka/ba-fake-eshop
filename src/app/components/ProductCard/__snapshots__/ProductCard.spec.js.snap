// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render correct snapshot 1`] = `"<div class=\\"ProductCard\\" id=\\"prodId\\"><form><label>Name:<input type=\\"text\\" name=\\"name\\" value=\\"\\"></label><input type=\\"submit\\" value=\\"Submit\\"></form><div class=\\"ProductCard__image-wrapper\\"><img src=\\"imgUrl\\" alt=\\"name product\\"></div><div class=\\"ProductCard__info\\"><div><h3>name</h3><p class=\\"ProductCard__info-price\\">1$</p></div><div class=\\"ProductCard__buttons-container\\"><button>Add to cart</button><button>Add to favourites</button><button>+</button><button>-</button><p>Quantity: 0</p></div></div></div>"`;
